src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -0,0 +1,24 @@
+package lotto.domain
+
+class Lotto(
+    lottoNumbers: List<LottoNumber>,
+) {
+    private val numbers = lottoNumbers.sorted()
+
+    val rawNumbers: List<Int>
+        get() = numbers.map { it.value }
+
+    init {
+        require(numbers.size == LOTTO_SIZE) { "Lotto must contain exactly 6 numbers." }
+    }
+
+    fun compareMatches(lotto: Lotto) = numbers.count { it in lotto.numbers }
```

Comment:
The book [[KR] The Essence of Object-Orientation](https://product.kyobobook.co.kr/detail/S000001628109) describes "message" as the following:
> One object accesses another by sending a message (message-transmission mechanism: the only way an object can access another object)
The fact that an object may receive a message means that the object may perform a responsibility corresponding to the message.
The outside object is naturally separated from the outside and the inside of the object because it can only be seen with respect to the message and not the inside of the object.
A message consists of a combination of recipient, message name, and factor.

You respected the object's responsibility and set a way to send a proper message. 👍